src/schema-manager.ts

```diff
@@ -210,6 +215,31 @@ export class PolygonSchema {
     }
   }
 
+  public async getAllSchemaByDID(did: string) {
+    try {
+      const isValidDid = validateDid(did)
+      if (!isValidDid) {
+        throw new Error('invalid did provided')
+      }
+      const didDetails = await this.resolver.resolve(did)
+
+      if (!didDetails?.didDocumentMetadata?.linkedResourceMetadata) {
+        throw new Error(
+          `The DID linked resource metadata for the given DID was not found!`,
+        )
+      }
```

Comment:
- I think by default if there are no schemas attached to the did instead of throwing error we should return empty array